packages/features/bookings/lib/handleCancelBooking.ts

```diff
@@ -180,6 +180,15 @@ async function handler(req: CustomRequest) {
     throw new HttpError({ statusCode: 400, message: "User not found" });
   }
 
+  const bookingFieldsArray =
+    (bookingToDelete?.eventType?.bookingFields as unknown as { name: string; required?: boolean }[]) || [];
```

Comment:
Please avoid type gymnastics here 😅. We have `eventTypeBookingFields` schema is `zod-utils`. You should use that instead so that you know properly what you are dealing with.